public/components/app.tsx

```diff
@@ -30,14 +52,117 @@ export const App = ({
+      [RAW_QUERY]: ''
+    }
+  });
+  const [queryResults, setQueryResults] = useState<ITabQueryResults>({
+    [initialTabId]: {}
+  });
+  const [fields, setFields] = useState<IExplorerTabFields>({
+    [initialTabId]: {
+      [SELECTED_FIELDS]: [],
+      [UNSELECTED_FIELDS]: []
+    }
+  });
+  const curQueriesRef = useRef(queries);
+  const [curSelectedTabId, setCurSelectedTab] = useState<string>(initialTabId);
+
+  function getTabId (prefix: string) {
+    return _.uniqueId(prefix);
+  }
+
+  const parentBreadcrumb = {
+    text: observabilityTitle,
+    href: 'observability#/'
+  };
+
   return (
-    <I18nProvider>
-      <>
-        <LogExplorer
-          http={ http }
-          plugins={ plugins }
-        />
-      </>
-    </I18nProvider>
+    <HashRouter>
+      <I18nProvider>
+        <>
+          <Switch>
+            <Route 
+              exact
+              path={['/', '/application_analytics', '/application_analytics/home']}
+              render={(props) => {
+                chrome.setBreadcrumbs([
+                  parentBreadcrumb,
+                  {
+                    text: 'Application analytics',
+                    href: 'observability#/application_analytics'
+                  },
+                ]);
+                return renderPageWithSidebar(<ApplicationAnalyticsHome />, 1);
+              } }
+            />
+            <Route
+              path={['/trace_analytics', '/trace_analytics/home']}
+              render={(props) => {
+                chrome.setBreadcrumbs([
+                  parentBreadcrumb,
+                  {
+                    text: 'Trace analytics',
+                    href: 'observability#/trace_analytics/home'
+                  },
+                ]);
+                return renderPageWithSidebar(<TraceAnalyticsHome />, 2) }
+              }
+            />
+            <Route 
+              exact
+              path={['/event/', '/event/home']}
+              render={(props) => {
+                chrome.setBreadcrumbs([
+                  parentBreadcrumb,
+                  {
+                    text: 'Event analytics',
+                    href: 'observability#/event/home'
+                  },
+                ]);
+                return renderPageWithSidebar(<EventExplorerHome />, 3);
+              } }
+            />
+            <Route
+              path={['/operational_panels', '/operational_panels/home']}
+              render={(props) => {
+                chrome.setBreadcrumbs([
+                  parentBreadcrumb,
+                  {
+                    text: 'Operational panels',
+                    href: 'observability#/operational_panels/home'
+                  },
+                ]);
+                return renderPageWithSidebar(<OperationalPanelsHome />, 4);
+              } }
+            />
+            <Route 
+              exact
+              path='/event/explorer'
+              render={(props) => <LogExplorer
+                http={ http }
```

Comment:
just a note, i initially passed `http` around in props in trace analytics because there were only three pages, but now it might be easier to put core dependencies to a context as many components will depend on it

Reply:
Makes sense, since we are going to create a lot of pages and components so it's good to pass the entire core dependencies. I'll change it in the next commit